unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -0,0 +1,248 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+//#cmake: static!
+//#test:device D400*
+//#test:device L500*
+
+
+#define CATCH_CONFIG_MAIN
+#include <stdlib.h> 
+#include "../../catch.h"
+#include "../../unit-tests-common.h"
+
+using namespace rs2;
+constexpr int RECEIVE_FRAMES_TIME = 10;
+
+TEST_CASE("Syncer dynamic FPS - throughput test", "[live]")
+{
+    typedef enum configuration
+    {
+        IR_ONLY,
+        IR_RGB_EXPOSURE,
+        STOP
+    }configuration;
+
+    class ir_cfg
+    {
+    public:
+        ir_cfg(sensor rgb_sensor,
+            sensor ir_sensor,
+            std::vector<rs2::stream_profile > rgb_stream_profile,
+            std::vector<rs2::stream_profile > ir_stream_profile) : _rgb_sensor(rgb_sensor), _ir_sensor(ir_sensor),
+            _rgb_stream_profile(rgb_stream_profile), _ir_stream_profile(ir_stream_profile) 
+        {
+            _delta = 0.0f;
+        }
+         
+        void test_configuration(configuration test)
+        {
+            std::cout << "==============================================" << std::endl;
+            std::string cfg = test == IR_ONLY ? "IR Only" : "IR + RGB";
+            std::cout << "Configuration: " << cfg << std::endl << std::endl;
+            
+            start_streaming(test);
+            process_validate_frames(test);
+            stop_streaming(test);
+            // analysis
+            analysis(test, _rgb_frames_arrival_info, "Color");
+            analysis(test, _ir_frames_arrival_info);
+        }
+
+    private:
+        void set_ir_exposure()
+        {
+            if (_ir_sensor.supports(RS2_OPTION_EXPOSURE))
+                REQUIRE_NOTHROW(_ir_sensor.set_option(RS2_OPTION_EXPOSURE, 18000)); // set exposure value to get fps = 30
```

Comment:
Pass in the actual exposure as function argument for flexibility ,e.g. repeating the test for 30/15 fps scenario.